// Copyright 2023 The XLS Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package xls;

import "xls/ir/channel.proto";

// Flags passed to the ir_converter_main binary.
//
// See ir_converter_options_flags.cc ABSL_FLAG() definitions for the meaning of
// these fields.
message IrConverterOptionsFlagsProto {
  optional string output_file = 1;
  optional string top = 2;
  optional string dslx_stdlib_path = 3;
  optional string dslx_path = 4;
  optional string package_name = 5;
  optional bool emit_fail_as_assert = 6;
  optional bool verify = 7;
  optional bool convert_tests = 8;
  optional string disable_warnings = 9;
  optional bool warnings_as_errors = 10;
  optional string interface_proto_file = 11;
  optional string interface_textproto_file = 12;
  optional FifoConfigProto default_fifo_config = 13;
  optional string enable_warnings = 14;
  optional bool proc_scoped_channels = 15;
  optional bool type_inference_v2 = 16;
  optional bool lower_to_proc_scoped_channels = 17;
}
